Wasm-container/README.md

```diff
@@ -0,0 +1,509 @@
+
+The above puts the `.cargo` and `.wasmer` directories into `$WASMHOME`.  Putting these into `$WASMHOME` instead of your home directory avoids interfering with any non-wasm Rust work you might be doing.  It also creates the `$HOME/WebAssembly/.env-setup` file.
+
+After you have used the above command to create your `$WASMHOME` toolbox, you can run the container interactively.
+
+## Launching an interactive shell to do Web Assembly development
+
+The shell-script `./vwasm-bash` starts an interactive shell inside the container.  
+
+When executing a shell inside the container, you need to set up your in-container PATH so the shell can find the `wasmer` and `rust` installations in the container.
+
+`wasmer` and rust installations are designed to be placed in a user's home directory.  This `copy-template-to-sandbox` command copied them to `$WASMHOME`.
+
+To set up your environment appropriately, run
+```shell
+source $WASMHOME/.env-setup
+```
+
+`vwasm-bash` has some defaults that may be overridden by setting environment variables. 
+
+| Name                      | Definition |
+|---------------------------|------------|
+| VERTICA_VERSION | The version number of the Vertica binary used in the build process. This value is optional for a [canonically-named Vertica binary](#building-with-a-canonically-named-vertica-binary).<br> You can use this variable to build containers for different Vertica versions. |
+| WASM_MOUNT | An optional space-separated list of host directories to mount in the UDx container --- that is, "I'm going to want to refer to files in these directories in, for example, my makefile, or as include directories in code to be compiled" |
+| VWASM_ENV | names a file filled with variable definitions.  This uses the --env-file option of the "docker run" command, so the file needs to be formatted for that option |
+
+If you have multiple vwasm containers, you can
+choose among them by specifying the tag, which
+is derived from `vwasmsdk:${OSTAG}-${VERTICA_VERSION}`, if those
+variables are set in the environment (you can also define `CONTAINER_TAG` if you've chosen a different container tagging scheme).
+
+`vwasm-bash` creates a shell with the following directories mounted, sharing the name they have in the host system:
+
+- `$HOME`
+- your currently connected directory
+- any directories listed in the environment variable `$VWASM_VOLUMES`
+
+In addition, a Docker volume is mounted as `/vwasmdata`.
+
+
+### An annoying thing about Rust
+
+The primary role of the container is to provide a place to install
+those components that require privileges to install.
+
+Rust and wasmer are designed to be installed privately, in one's
+home directory.
+
+While we play games in the Dockerfile with installing those tools in
+`/usr/WebAssembly/template`, and provide a script to copy them into
+one's home directory, the changes don't always seem to stick between
+container invocations.  This manifests for me as the following:
```

Comment:
Since this is a public repo not a blog, maybe "This can manifest as the following" and then "when you try to use..."?